security/token-file-provider.1.rst

```diff
@@ -0,0 +1,164 @@
+NAME
+====
+token-file-provider – Generate Vault tokens for EdgeX services
+
+
+SYNOPSIS
+========
+token-file-provider [-h|-help] [-confdir confdir]
+
+
+DESCRIPTION
+===========
+token-file-provider generates per-service Vault tokens for EdgeX services
+so that they can make authenticated connections to Vault to retrieve
+application secrets.
+token-file-provider implements a generic secret seeding mechanism based
+on pre-created files and is designed for maximum portability.
+token-file-provider takes a configuration file that specifies the services
+for which tokens shall be generated and the Vault access policy
+that shall be applied to those tokens.
+token-file-provider assumes that there is some underlying protection mechanism
+that will be used to prevent EdgeX services from reading each other’s tokens.
+
+
+OPTIONS
+=======
+  \-h
+    Display help text
+
+  \-help
+    Display help text
+
+  \-confdir confdir
+    Look in this directory for configuration.toml instead.
+
+
+FILES
+=====
+
+configuration.toml
+------------------
+This file specifies the TCP/IP location of the Vault service
+and parameters used for Vault token generation.
+
+::
+
+  [SecretService]
+  Server = "localhost"
+  Port = 8200 
+
+  [TokenFileProvider]
+  PrivilegedTokenPath = /run/edgex/secrets/token-file-provider/vault-token.json
+  ConfigFile = token-config.json
+  OutputDir = /run/edgex/secrets/
+  OutputFilename = vault-token.json
+
+
+vault-token.json
+----------------
+This file contains a token use to authenticate to Vault.
+
+::
+
+  {
+    "auth": {
+      "client_token": "s.wOrq9dO9kzOcuvB06CMviJhZ"
+    }
+  }
+
+
+token-config.json
+-----------------
+This configuration file tells token-file-provider which tokens to generate.
+
+::
+
+  {
+    "service-name": {
+      "edgex-use-defaults": true,
+      "custom_policy": [
+        {
+          "path": {
+            "secret/edgex/pki/tls/edgex-kong": {
+              "capabilities": [ "list", "read" ]
+            }
+          }
+        }
+      ],
+      "custom_token_parameters": { }
+    }
+  }
+
+
+When edgex-use-default is true (the default),
+the following is appended to the policy array for the generated token:
```

Comment:
should this say 

```suggestion
the following is appended to the policy's `capabilities` array for the generated token:
```
?

Reply:
https://github.com/edgexfoundry/edgex-docs/pull/35/commits/bd99b6ff553c24070f5265d0c5914089d399ff3f